ui/src/app/settings/components/project-details/project-details.tsx

```diff
@@ -455,616 +455,790 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
+                                                                field={`spec.namespaceResourceBlacklist[${i}].group`}
+                                                                component={AutocompleteField}
+                                                                componentProps={{items: Groups}}
+                                                            />
+                                                        </div>
+                                                        <i
+                                                            className='fa fa-times'
+                                                            onClick={() =>
+                                                                formApi.setValue('spec.namespaceResourceBlacklist', removeEl(formApi.values.spec.namespaceResourceBlacklist, i))
+                                                            }
+                                                        />
+                                                    </div>
+                                                ))}
+                                                <button
+                                                    className='argo-button argo-button--short'
+                                                    onClick={() =>
+                                                        formApi.setValue(
+                                                            'spec.namespaceResourceBlacklist',
+                                                            (formApi.values.spec.namespaceResourceBlacklist || []).concat({
+                                                                group: '*',
+                                                                kind: '*'
+                                                            })
+                                                        )
+                                                    }>
+                                                    ADD RESOURCE
+                                                </button>
+                                            </React.Fragment>
+                                        )}
+                                    </DataLoader>
+                                )}
+                                items={[]}
+                            />
+
+                            <div className='white-box editable-panel'>
+                                <div className='white-box__details'>
+                                    <p>
+                                        NAMESPACE RESOURCE DENY LIST FROM GLOBAL PROJECT
+                                        {helpTip('Namespace-scoped K8s API Groups and Kinds which are prohibited from being deployed from global project')}
+                                    </p>
+                                    <React.Fragment>
+                                        {gpNamespaceResourceBlacklist.length > 0 ? (
```

Comment:
Let's hide this section is it has no rows. And same for other inherited lists from global project.